teacher.rb

```diff
@@ -0,0 +1,13 @@
+require_relative './person'
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'unknown', parent_permission: true)
+    super(id, name, age, parent_permission)
```

Comment:
- [ ] Kindly remove `id` from the list of parameters here.

- [ ] In the `super` parameters kindly re-arrange them, `age` should come before `name` based on how the Person constructor is set up. Also ensure the `parent_permission` is assigned properly.

  ```
  super(age, name, parent_permission: parent_permission)
  ```